src/ck-core/ckcallback.C

```diff
@@ -367,69 +372,68 @@ void CkCallback::send(void *msg) const
 	case sendChare: //Send message to a chare
 		if (!msg) msg=CkAllocSysMsg();
                 if (d.chare.hasRefnum) CkSetRefNum(msg, d.chare.refnum);
-		CkSendMsg(d.chare.ep, msg, &d.chare.id);
+		CkSendMsg(d.chare.ep, msg, &d.chare.id, opts);
 		break;
 	case isendChare: //inline send-to-chare
 		if (!msg) msg=CkAllocSysMsg();
                 if (d.chare.hasRefnum) CkSetRefNum(msg, d.chare.refnum);
-		CkSendMsgInline(d.chare.ep, msg, &d.chare.id);
+		CkSendMsgInline(d.chare.ep, msg, &d.chare.id, opts);
 		break;
 	case sendGroup: //Send message to a group element
 		if (!msg) msg=CkAllocSysMsg();
                 if (d.group.hasRefnum) CkSetRefNum(msg, d.group.refnum);
-		CkSendMsgBranch(d.group.ep, msg, d.group.onPE, d.group.id);
+		CkSendMsgBranch(d.group.ep, msg, d.group.onPE, d.group.id, opts);
 		break;
 	case sendNodeGroup: //Send message to a group element
 		if (!msg) msg=CkAllocSysMsg();
                 if (d.group.hasRefnum) CkSetRefNum(msg, d.group.refnum);
-                if (_entryTable[d.group.ep]->isImmediate) opts = CK_MSG_IMMEDIATE;
 		CkSendMsgNodeBranch(d.group.ep, msg, d.group.onPE, d.group.id, opts);
 		break;
 	case isendGroup: //inline send-to-group element
 		if (!msg) msg=CkAllocSysMsg();
                 if (d.group.hasRefnum) CkSetRefNum(msg, d.group.refnum);
-		CkSendMsgBranchInline(d.group.ep, msg, d.group.onPE, d.group.id);
+		CkSendMsgBranchInline(d.group.ep, msg, d.group.onPE, d.group.id, opts);
 		break;
 	case isendNodeGroup: //inline send-to-group element
 		if (!msg) msg=CkAllocSysMsg();
                 if (d.group.hasRefnum) CkSetRefNum(msg, d.group.refnum);
-                if (_entryTable[d.group.ep]->isImmediate) opts = CK_MSG_IMMEDIATE;
 		CkSendMsgNodeBranchInline(d.group.ep, msg, d.group.onPE, d.group.id, opts);
 		break;
 	case sendArray: //Send message to an array element
 		if (!msg) msg=CkAllocSysMsg();
                 if (d.array.hasRefnum) CkSetRefNum(msg, d.array.refnum);
-		CkSetMsgArrayIfNotThere(msg);
-		CkSendMsgArray(d.array.ep, msg, d.array.id, d.array.idx.asChild());
+		CkSetMsgArrayIfNotThere(msg, policy);
+		CkSendMsgArray(d.array.ep, msg, d.array.id, d.array.idx.asChild(), opts);
 		break;
 	case isendArray: //inline send-to-array element
 		if (!msg) msg=CkAllocSysMsg();
                 if (d.array.hasRefnum) CkSetRefNum(msg, d.array.refnum);
-		CkSendMsgArrayInline(d.array.ep, msg, d.array.id, d.array.idx.asChild());
+		CkSetMsgArrayIfNotThere(msg, policy);
```

Comment:
Why does this (and the `bcastArray` case below) now set the policy where it previously didn't? Should it have been there in the past for inline and broadcast sends but it was just never added here?

Reply:
Unsure, could not determine a reason; seems like a curious omission.